hibernate-core/src/main/java/org/hibernate/annotations/SqlTypeCode.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.annotations;
+
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * @asciidoc
+ *
+ * Allows specifying the SQL type code to use for a column mapping.  Must
+ * resolve to a {@link org.hibernate.type.descriptor.sql.SqlTypeDescriptor}.
+ *
+ * The SQL type code is defined by a {@linkplain java.sql.Types JDBC Types} code
+ * or a custom code.
+ *
+ * ````
+ * @Entity
+ * class User {
+ *     ...
+ *     // By default Hibernate maps Java's Integer to JDBC's INTEGER
+ *     // but here we want to use JDBC's TINYINT instead.
+ *     @SqlType( Types.TINYINT )
+ *     int getAge() { ... }
```

Comment:
I noticed that after the latest gradle upgrade in order to avoid checkstyle errors a space is needed between `@SqlType` and the `(` in the javadoc

Reply:
may be I'm wrong but it seems this doc should belong to `hibernate-core/src/main/java/org/hibernate/annotations/SqlType.java`

Reply:
Well actually, these do belong here.  I just used the wrong annotation in the comment.  Still you are correct that comments are not correct atm.